### DISCUSSION TITLE
Federated Gateway implementation using Golang instead of Apollo Node.js?

### DISCUSSION THREAD

**Discussion by arkmech** [99designs/gqlgen#2425] [General]
Docs: https://gqlgen.com/recipes/federation/

The Federated Gateway is just using Apollo on a Node.js server. One of the main reasons I would choose Go is for performance, but looks like using Node.js as Federated Gateway, Node.js becomes the bottleneck.

Is there a Golang implementation for Federated Gateway?

**Comment by argoyle** [comment]
I use a variant of the example from here: https://github.com/wundergraph/graphql-go-tools

It even supports subscriptions.

**Reply by manfe** [reply]
@argoyle In this file: https://github.com/wundergraph/graphql-go-tools/blob/master/examples/federation/gateway/main.go

it have:

```
// It's just a simple example of graphql federation gateway server, it's NOT a production ready code.
```

What would be necessary to have a production ready code? Or what would be necessary to change to achieve it?

**Reply by argoyle** [reply]
I guess that depends on your requirements for production ready. I've made minor changes here and there to suit my own code style etc but no major changes.

**Comment by chadxzs** [comment]
For what it's worth, Apollo now has [apollo-router](https://github.com/apollographql/router) which is a federation router built in Rust. It does not support subscriptions at this time, though.
